### PR TITLE
Move DelaySpanBugEmitted to ty::context

### PR DESCRIPTION
**Pull request by mark-i-m**
This makes it even hard to abuse.

r? @eddyb 

cc @LeSeulArtichaut as this will probably conflict with your PR :/

### CODE REVIEW

**Comment by mark-i-m** [comment]
Ping @eddyb. I think this should be a relatively easy PR to review.

**Review by eddyb — APPROVED** [review]
r=me, please PM on Zulip if I lose track of PRs

File: src/librustc_middle/ty/sty.rs [hunk 1]
```diff
@@ -1793,7 +1791,11 @@ impl<'tcx> TyS<'tcx> {
 
     #[inline]
     pub fn is_phantom_data(&self) -> bool {
-        if let Adt(def, _) = self.kind { def.is_phantom_data() } else { false }
+        if let Adt(def, _) = self.kind {
+            def.is_phantom_data()
+        } else {
+            false
+        }
```
Comment (eddyb) [inline src/librustc_middle/ty/sty.rs:1798]: This (and the other similar change) don't look right, can you try `./x.py fmt`? You may have formatted them with a newer version of `rustfmt`.

Comment (mark-i-m) [reply src/librustc_middle/ty/sty.rs:1798]: Good catch! Fixed now.

**Review by eddyb — APPROVED** [review]

**Pull request merged**